### PR TITLE
add patch client tool

### PR DESCRIPTION
**Pull request by allanon**
- [ ] Code Review
- [ ] QA Review

Add patch client. This requires `grf_extract`, which is a command line GRF extraction tool I wrote based on libgrf. The intended purpose of this tool is to help keep the official files up to date.

`grf_extract` is in my fork of `libgrf`: https://github.com/OpenKore/grf

### CODE REVIEW

File: tools/patch.pl [hunk 1]
```diff
@@ -0,0 +1,305 @@
+#!/usr/bin/perl -w
+
+use strict;
+use warnings;
+
+use Date::Manip;
```
Comment (allanon) [inline tools/patch.pl]: Is `Date::Manip` used? I don't see any references to `UnixDate`.

**Comment by kaliwanagan** [comment]
@allanon I know it's been over a year now, but are you still interested in merging this?

**Comment by allanon** [comment]
Yes, certainly, I use this tool every time I make one of my automatic iRO updates.

**Comment by allanon** [comment]
@kaliwanagan I've forked the `grf_extract` repo here, and I'll keep it up to date (if there is ever anything to change) here instead of on my github.

I also switched the whitespace to openkore standards (ie, tabs). :)

**Comment by kaliwanagan** [comment]
:+1:

**Pull request merged**